profiles/bicycle.lua

```diff
@@ -485,9 +489,21 @@ function way_function (way, result)
   end
 
 
+
   -- maxspeed
   limit( result, maxspeed, maxspeed_forward, maxspeed_backward )
 
+
+  -- not routable if no speed assigned
+  -- this avoid assertions in debug builds
+  if result.forward_speed <= 0 then
```

Comment:
i fully agree, it seems very akward. i'm a bit puzzled that this problem didn't appear earlier, i would think setting the mode and not the speed would happen in other cases too, but appearently not.

not sure i understand your suggestion. but we would remove the assertion and instead silently ignore ways that have neither speed not duration.